lib/rutema/core/engine.rb

```diff
@@ -128,17 +192,41 @@ def run_test specification
       end
       return status
     end
+
+    ##
+    # Instantiate a new class of a given type passing it a given configuration
+    # upon construction
+    #
+    # * +definition+ - class of which a new instance shall be instantiated
+    # * +configuration+ - Configuration instance which shall be passed to the
+    #   initializer of the to be created instance
     def instantiate_class definition,configuration
       if definition[:class]
         klass=definition[:class]
         return klass.new(configuration)
       end
       return nil
     end
+
+    ##
+    # Check if the given test identifier belongs to the normal test cases or to
+    # one of the special ones (the test (suite) setups and teardowns)
+    #
+    # * +test_identifier+ - the test identifier which shall be checked if it
+    #   belongs to the normal test cases or to the special ones
     def is_spec_included? test_identifier
       full_path=File.expand_path(test_identifier)
       return @configuration.tests.include?(full_path) || is_special?(test_identifier) 
     end
+
+    ##
+    # Check if the given test identifier is a (suite) setup or teardown one
+    #
+    # This checks if the given identifier was configured as (suite) setup or
+    # teardown within the rutema run's configuration.
+    #
+    # * +test_identifier+ - the test identifier which shall be checked for being
```

Comment:
see above

Reply:
I'm sorry, but here the context is different. How could or should this one be improved.

The `test_identifier` name is anyhow one I'm not satisfied yet. I would rename this during the refactoring itself. This pull request is more about laying a baseline for the refactoring and getting my head around the code.